modules/phase_field/src/ics/IsolatedBoundingBoxIC.C

```diff
@@ -0,0 +1,133 @@
+#include "MooseMesh.h"
+#include <iostream>
+
+registerMooseObject("PhaseFieldApp", IsolatedBoundingBoxIC);
+
+namespace sizeVector_def_isolated
+{
+// Convenience function for sizing a vector to "n" given a vector with size 1 or "n"
+std::vector<Real>
+sizeVector(std::vector<Real> v, std::size_t size)
+{
+  if (v.size() == 1)
+    return std::vector<Real>(size, v[0]);
+  else
+    return v;
+}
+}
+
+template <>
+InputParameters
+validParams<IsolatedBoundingBoxIC>()
+{
+  InputParameters params = validParams<InitialCondition>();
+  params.addClassDescription(
+      "Specify variable values inside and outside a list of isolated boxes shaped "
+      "axis-aligned regions defined by pairs of opposing corners");
+  params.addRequiredParam<std::vector<Point>>(
+      "smaller_coordinate_corners",
+      "For 1D, these are the left points; for 2D, these are the bottom left corners; for 3D, these "
+      "are the back bottom left corners. The format is (corner1_x corner1_y corner1_z corner2_x "
+      "corner2_y corner2_z ...)");
+  params.addRequiredParam<std::vector<Point>>(
+      "larger_coordinate_corners",
+      "For 1D, these are the right points; for 2D, these are the top right corners; for 3D, these "
+      "are the front top right corners. The format is (corner1_x corner1_y corner1_z corner2_x "
+      "corner2_y corner2_z ...)");
+  params.addRequiredParam<std::vector<Real>>("inside",
+                                             "The value of the variable inside each box "
+                                             "(one value per box)");
+
+  params.addParam<Real>("outside", 0.0, "The value of the variable outside the boxes");
+
+  params.addParam<Real>(
+      "int_width", 0.0, "The width of the diffuse interface. Set to 0 for sharp interface.");
+
+  params.addClassDescription("Allows setting the initial condition of the value of a field inside "
+                             "multiple isolated bounding boxes.");
+  return params;
+}
+
+IsolatedBoundingBoxIC::IsolatedBoundingBoxIC(const InputParameters & parameters)
+  : InitialCondition(parameters),
+    _c1(getParam<std::vector<Point>>("smaller_coordinate_corners")),
+    _c2(getParam<std::vector<Point>>("larger_coordinate_corners")),
+    _nbox(_c1.size()),
+    _int_width(getParam<Real>("int_width")),
+    _dim(_fe_problem.mesh().dimension()),
+    _inside(sizeVector_def_isolated::sizeVector(getParam<std::vector<Real>>("inside"), _nbox)),
+    _outside(getParam<Real>("outside"))
+{
+  // make sure inputs are the same length
+  if (_c2.size() != _nbox || _inside.size() != _nbox)
+    mooseError("vector inputs must all be the same size");
+}
+
+Real
+IsolatedBoundingBoxIC::value(const Point & p)
+{
+  Real value = _outside; // if p is not inside the largest box, assign the last element of inside
+                         // which is 'outside' to value
+
+  if (_int_width < 0.0)
+    mooseError("'int_width' should be non-negative");
+
+  if (_int_width == 0.0)
```

Comment:
So in this case, isn't this exactly what MultiBoundBoxIC does?

Reply:
MultiBoundingBoxIC does not have int_width. I don't think this piece of code exists in MultiBoundingBoxIC.

Reply:
Right, but there's no reason you can't add it. You could define the default width to be 0 then everybody already using that object won't need to make any changes, but they can just add a parameter to get a diffuse interface without swapping it out. It also cuts down on the number of objects and code duplication.

Reply:
If someone is already using MultiBoundingBoxIC, they don't need to make any change since I won't change how that IC works (other than deleting the helper function and use .assign method). But since I haven't found an easy algorithm to achieve a diffused interface in the current MultiBoundingBoxIC object, we can't just add an int_width parameter to the current object, if that's what you meant. That's why I made the two new objects. Please let me know if it doesn't make sense.